libs/community/langchain_community/vectorstores/pinecone.py

```diff
@@ -369,9 +360,7 @@ def get_pinecone_index(
                 "Please install it with `pip install pinecone-client`."
             )
 
-        pinecone_client_version = pinecone.__version__
-
-        if version.parse(pinecone_client_version) >= version.parse("3.0.0.dev"):
```

Comment:
will this work with old pinecone clients? If not, it's probably worth continuing to check version number because community packages don't offer guarantees about which package version is installed.